sound/soc/sof/sof-audio.c

```diff
@@ -556,27 +635,22 @@ static int sof_audio_probe(struct platform_device *pdev)
 	INIT_LIST_HEAD(&sof_audio->dai_list);
 	INIT_LIST_HEAD(&sof_audio->route_list);
 
+	sof_audio->platform = dev_name(&pdev->dev);
+
 	audio_client->client_data = sof_audio;
 
 	/* set IPC RX callback */
 	audio_client->sof_client_rx_cb = sof_audio_rx_message;
 
 	snd_sof_ipc_rx_register(audio_client, &pdev->dev);
 
-	/* check machine info */
-	ret = sof_machine_check(pdev);
-	if (ret < 0) {
-		dev_err(&pdev->dev, "error: failed to get machine info %d\n",
-			ret);
+	/* select machine driver */
+	ret = sof_audio_select_machine(pdev, desc);
+	if (ret < 0)
 		return ret;
```

Comment:
@ranj063 because machine driver for IMX will be probed separately by placing a description in DT I think all the functions related to machine drivers in sof-audio.c should be no-ops.

Sorry, for misleading you. Will try to send some patches on top of this PR.

Not sure we even need the sof_fw_mach structure now.

There are still some things I need to figure out:
* if machine driver probing is decoupled from SOF core how do we set .ignore machine field?
* platform_device_register_data that creates machine device for ACPI / PCI should be abstracted as a per device ops, with no-op for IMX

Reply:
@dbaluta no worries. this PR is still under review. So please let me know when you have more info.

Reply:
@ranj063 I have a wild proposal for you. What do you think if I base my patches on current sof-dev branch without this PR, then you rebase your patches on mine.

Thus it would be much easier for me to backport support for IMX machine driver in 5.4

This is just a thought.

Reply:
@dbaluta I think thats better. I will undo the common sof_fw_mach changes in this series and wait for your PR

Reply:
great, thanks!